integration/experiment/ffnet/gen_neural_net.py

```diff
@@ -0,0 +1,277 @@
+            ['MSR::PPERF:PCNT-package-0', 'MSR::APERF:ACNT-package-0'],
+            ],
+            'gpu':[]}
+
+    if type(input_list) == str:
+        input_list = [input_list]
+
+    for inp in input_list:
+        #TODO: Check if hdf, if not, convert
+        df = pd.read_hdf(inp)
+        if 'app-config' not in  df:
+            sys.stderr.write('<geopm> Error: No app-config in input data. Have you used gen_hdf_from_fsweep.py to create this HDF?\n')
+            sys.exit(1)
+
+        df["region-id"] = df["app-config"]
+
+        #TODO: Confirm we can delete the following
+        # for parres, the region hash can't be determined by GEOPM
+        #df = df[~df['REGION_HASH'].isna()]
+        #df = df[~df['REGION_HASH'].isin(region_ignore)]
+        #df["region_id"] = df["REGION_HASH"] + suffix
+
+        dfs.append(df)
+
+    df_traces = pd.concat(dfs)
+
+    domains_to_train = []
+
+# Check traces to determine which domains we can train on
+    for domain in X_columns_domain:
+        add_domain = True
+        for signal in set(X_columns_domain[domain] + sum(ratios_domain[domain], [])):
+            if signal not in df_traces.columns:
+                print (f"Missing signal {signal} is required for training {domain}.")
+                add_domain = False
+                break
+        if add_domain:
+            print (f"Training on {domain}")
+            domains_to_train.append(domain)
+
+    if len(domains_to_train) == 0:
+        sys.stderr.write('<geopm> Error: No training domain is complete\n')
+        sys.exit(1)
+
+    print("Training to identify these regions:")
+    region_ids = sorted(list(df_traces["region-id"].unique()))
+    print(", ".join(region_ids))
+    mapping = {region_id: region_ids.index(region_id) for region_id in region_ids}
+    df_traces["region-id"] = df_traces['region-id'].map(mapping)
+
+    for domain in domains_to_train:
+        for num,den in ratios_domain[domain]:
+            name = 'delta_{}/delta_{}'.format(num, den)
+            X_columns_domain[domain].append(name)
+            df_traces[name] = df_traces[num].diff() / df_traces[den].diff()
+
+        df_traces.replace([np.inf, -np.inf], np.nan, inplace=True)
+        is_missing_data = df_traces[X_columns_domain[domain] + y_columns].isna().sum(axis=1) > 0
+        df_traces_domain = df_traces.loc[~is_missing_data]
+
+        model = train_model(df_traces_domain, X_columns_domain[domain], y_columns)
+
+        model_scripted = torch.jit.script(model)
+        model_scripted.save(f"{output_name}_{domain}.pt")
+
+        model_out = open(f"{output_name}_{domain}.json", "w")
+        model.train = False
+        with torch.no_grad():
+            json.dump(model_to_json(model, X_columns_domain[domain], ["trace_" + region_id  for region_id in region_ids], describe_net), model_out)
+        model_out.close()
+
+
+if __name__ == '__main__':
+
+     parser = argparse.ArgumentParser()
+     parser.add_argument('--output',
```

Comment:
Minor suggestion (applies in other places/files as well):

```suggestion
     parser.add_argument('--output-prefix',
```